src/languages/dos.js

```diff
@@ -12,11 +12,30 @@ export default function(hljs) {
     /^\s*@?rem\b/, /$/,
     { relevance: 10 }
   );
+
+  // for matching comments starting with ::
+  const COMMENT_2 = hljs.COMMENT(
+    /^::.*$/, /$/,
+    { relevance: 10 }
+  );
   const LABEL = {
     className: 'symbol',
-    begin: '^\\s*[A-Za-z._?][A-Za-z0-9_$#@~.?]*(:|\\s+label)',
+    begin: '^:[A-Za-z._?][A-Za-z0-9_$#@~.?]*',
+    relevance: 0
+  };
+
+  const DISK_CHANGE = {
+    className: 'symbol',
+    begin: '^[A-Za-z]:\\?$',
+    relevance: 0
+  };
+
+  const OUTPUT_REDIRECT = {
+    className: 'symbol',
+    begin: '[1-2]?[>]>{1}\s*[^&\s]+',
```

Comment:
Please add a markup test or two for this so I can understand what exactly you're trying to match.  This looks a bit overly complex to me, but maybe I don't understand the goal.